TileStache/Goodies/VecTiles/transform.py

```diff
@@ -1181,12 +1185,86 @@ def _make_new_properties(props, props_instructions):
 
     return new_props
 
+
+def _snap_to_grid(shape, grid_size):
+    """
+    Snap coordinates of a shape to a multiple of `grid_size`.
+
+    This can be useful when there's some error in point
+    positions, but we're using an algorithm which is very
+    sensitive to coordinate exactness. For example, when
+    calculating the boundary of several items, it makes a
+    big difference whether the shapes touch or there's a
+    very small gap between them.
+
+    This is implemented here because it doesn't exist in
+    GEOS or Shapely. It exists in PostGIS, but only because
+    it's implemented there as well. Seems like it would be a
+    useful thing to have in GEOS, though.
+
+    >>> _snap_to_grid(Point(0.5, 0.5), 1).wkt
+    'POINT (1 1)'
+    >>> _snap_to_grid(Point(0.1, 0.1), 1).wkt
+    'POINT (0 0)'
+    >>> _snap_to_grid(Point(-0.1, -0.1), 1).wkt
+    'POINT (-0 -0)'
+    >>> _snap_to_grid(LineString([(1.1,1.1),(1.9,0.9)]), 1).wkt
+    'LINESTRING (1 1, 2 1)'
+    _snap_to_grid(Polygon([(0.1,0.1),(3.1,0.1),(3.1,3.1),(0.1,3.1),(0.1,0.1)],[[(1.1,0.9),(1.1,1.9),(2.1,1.9),(2.1,0.9),(1.1,0.9)]]), 1).wkt
+    'POLYGON ((0 0, 3 0, 3 3, 0 3, 0 0), (1 1, 1 2, 2 2, 2 1, 1 1))'
+    >>> _snap_to_grid(MultiPoint([Point(0.1, 0.1), Point(0.9, 0.9)]), 1).wkt
+    'MULTIPOINT (0 0, 1 1)'
+    >>> _snap_to_grid(MultiLineString([LineString([(0.1, 0.1), (0.9, 0.9)]), LineString([(0.9, 0.1),(0.1,0.9)])]), 1).wkt
+    'MULTILINESTRING ((0 0, 1 1), (1 0, 0 1))'
+    """
+
+    # snap a single coordinate value
+    def _snap(c):
+        return grid_size * round(c / grid_size, 0)
+
+    # snap all coordinate pairs in something iterable
+    def _snap_coords(c):
+        return [(_snap(x), _snap(y)) for x, y in c]
+
+    # recursively snap all coordinates in an iterable over
+    # geometries.
+    def _snap_multi(geoms):
+        return [_snap_to_grid(g, grid_size) for g in geoms]
+
+    shape_type = shape.geom_type
+    if shape_type == 'Point':
+        return Point(_snap(shape.x), _snap(shape.y))
+
+    elif shape_type == 'LineString':
+        return LineString(_snap_coords(shape.coords))
+
+    elif shape_type == 'Polygon':
+        exterior = LinearRing(_snap_coords(shape.exterior.coords))
+        interiors = []
+        for interior in shape.interiors:
+            interiors.append(LinearRing(_snap_coords(interior.coords)))
+        return Polygon(exterior, interiors)
+
+    elif shape_type == 'MultiPoint':
+        return MultiPoint(_snap_multi(shape.geoms))
+
+    elif shape_type == 'MultiLineString':
+        return MultiLineString(_snap_multi(shape.geoms))
+
+    elif shape_type == 'MultiPolygon':
+        return MultiPolygon(_snap_multi(shape.geoms))
```

Comment:
There's a couple of issues with GeometryCollections - first, we shouldn't have them at the point this function is called, as there's a check first to see if it's a Polygon or MultiPolygon. I did write the code to be generic, but... Second, found that in v1.4.3 of Shapely, there's [no way to instantiate a GeometryCollection with geometries](https://github.com/Toblerity/Shapely/blob/1.4.3/shapely/geometry/collection.py#L18). I would create it and then set `geoms`, but I got `http://lists.gispython.org/pipermail/community/2012-November/003156.html`. It seems that it's [intentionally not supported](http://lists.gispython.org/pipermail/community/2012-November/003156.html).